aroquev/src/main/webapp/script.js

```diff
@@ -62,25 +62,108 @@ function showPreviousSlide() {
 
 // Thumbnail image controls
 function updateCurrentSlide(n) {
-  slideIndex = n;
-  showSlides(slideIndex);
+	slideIndex = n;
+	showSlides(slideIndex);
 }
 
 function showSlides(n) {
-  var slides = document.getElementsByClassName("mySlides");
+	var slides = document.getElementsByClassName("mySlides");
 	var dots = document.getElementsByClassName("dot");
-  slideIndex = n % slides.length;
-  if (slideIndex == -1) {
-    slideIndex = slides.length - 1;
-  }
-  for (const slide of slides) {
-    slide.style.display = "none";
-  }
-  for (const dot of dots) {
-      dot.classList.remove("active");
-  }
-  slides[slideIndex].style.display = "block";
-  dots[slideIndex].classList.add("active");
+	slideIndex = n % slides.length;
+	if (slideIndex == -1) {
+		slideIndex = slides.length - 1;
+	}
+	for (const slide of slides) {
+		slide.style.display = "none";
+	}
+	for (const dot of dots) {
+		dot.classList.remove("active");
+	}
+	slides[slideIndex].style.display = "block";
+	dots[slideIndex].classList.add("active");
 }
 
 initSlides();
+
+/**
+ * Adds a random car and its data to the page.
+ */
+function getRandomCar() {
+	const cars = [
+		{
+			name: "McLaren 720S",
+			power: "720 hp",
+			country: "England",
+			funFact: "Just look at it, it is beautiful!",
+			path: "images/720s.jpg"
+		},
+		{
+			name: "Porsche 918 Spyder",
+			power: "720 hp",
+			country: "Germany",
+			funFact: "Hybrid power for the win!",
+			path: "images/porsche.jpeg"
+		},
+		{
+			name: "Ferrari FXX Evoluzione",
+			power: "848 hp",
+			country: "Italy",
+			funFact: "You should really listen to the sound this one makes",
+			path: "images/ferrari.jpg"
+		},
+		{
+			name: "Lamborghini Sesto Elemento",
+			power: "562 hp",
+			country: "Italy",
+			funFact: "The whole thing weights less that a tonne thanks to carbon fiber!",
+			path: "images/lambo.jpg"
+		},
+		{
+			name: "Mercedes-AMG G63 6x6",
+			power: "536 hp",
+			country: "Germany",
+			funFact: "Taking offroading to a whole new level.",
+			path: "images/mercedes.jpeg"
+		},
+		{
+			name: "Subaru WRX STI",
+			power: "536 hp",
+			country: "Japan",
+			funFact: "Boxster engines rock!",
+			path: "images/subaru.jpg"
+		},
+		{
+			name: "Mini JCW",
+			power: "306 hp",
+			country: "England",
+			funFact: "I currently own a Mini :).",
+			path: "images/mini.jpeg"
+		},
+		{
+			name: "Volkswagen Sedan",
+			power: "Unlimited hp",
+			country: "Germany",
+			funFact: "Grandpa's car <3, learned to drive in it.",
+			path: "images/vw.jpg"
+		}
+	];
+
+	// Pick a random car.
+	const car = cars[Math.floor(Math.random() * cars.length)];
+
+	// Add it to the page.
```

Comment:
You should construct the whole DOM structure for a car in JS and attach it to the parent car view container rather than swapping out the text of several elements. This way the JS is not coupled with the HTML. If you wanted to change/add an attribute of car you would only have to do it in one place.

For example

```js
buildCarView(car) {
  const carView = document.createElement('div');

  const image = document.createElement('img');
  image.src = car.path;
  image.classList.add('car-image');

  const name = document.createElement('div');
  name.textContent = `Name: ${car.name}`;

  // More elements...

  carView.appendChild(image);
  // append stuff to carView

  return carView;
}
```

Inside of `getRandomCar()`
```js
const carContainer = document.getElementById('car-container');
const carView = buildCarView(car);
// replace contents of carContainer with carView
```

Reply:
FWIW it is also waaaay faster for the browser to construct DOM nodes in JS rather than perform updates bit-by-bit.

Reply:
Wow, nice! Done.